easyrsa3/easyrsa

```diff
@@ -838,6 +839,7 @@ revoke() {
 Error: didn't find a file base name as the first argument.
 Run easyrsa without commands for usage and command help."
 	crt_in="$EASYRSA_PKI/issued/$1.crt"
+	key_in="$EASYRSA_PKI/private/$1.key"
```

Comment:
Same as above, this can be reused later if a client or consumer loses it.

Reply:
The user loses the certificate but not the key? If it loses the key, I would prefer to regenerate both (for security reasons).

The general recommendation is to discard CSR when it is used. IMHO, a software should encourage good practices. The certificate expirity is there to force an update on key pair and ciphers. If you intend to reuse it, simply use a larger "expire date". BTW, easyrsa reduced that recently, which goes against the logic of keeping CSR/privkey.

The only reason I can think (but not recommend) to keep req/privkey is when you need only to renew an existing certificate.

There is two scenarios here:
1) if you have client privkey, you do not need csr as you can regenerate it from existing certificate
2) if you don't have client privkey, you might want to reuse old CSR, but it is not recommended.

In 1) you would normally share the keypair+certificates in a PKCS#12. I woundn't create a specific procedure to import only the new certificate. As the user is importing both, simply regenerate the whole thing.
In 2), the user generated the CSR. The user can resubmit the same CSR but I would still recommend that it generates a new pair and a new CSR.